File: apphub-frontend/src/common/js/dao/endpoints/StorageEndpoints.js
import { Endpoint, RequestMethod } from "../dao";

export const STORAGE_UPLOAD_FILE = new Endpoint(RequestMethod.PUT, "/api/storage/{storedFileId}");
export const STORAGE_DOWNLOAD_FILE = new Endpoint(RequestMethod.GET, "/api/storage/{storedFileId}");
export const STORAGE_GET_METADATA = new Endpoint(RequestMethod.GET, "/api/storage/{storedFileId}/metadata");

File: apphub-lib/apphub-lib-config/apphub-lib-config-common/src/main/java/com/github/saphyra/apphub/lib/config/common/endpoints/StorageEndpoints.java
package com.github.saphyra.apphub.lib.config.common.endpoints;

public class StorageEndpoints {
    public static final String EVENT_CLEAN_UP_STORED_FILES = "/event/storage/cleanup/stored-files";
    public static final String EVENT_FILE_CLEANUP = "/api/storage/cleanup/files";

    public static final String STORAGE_UPLOAD_FILE = "/api/storage/{storedFileId}";
    public static final String STORAGE_INTERNAL_DELETE_FILE = "/internal/storage/{storedFileId}";
    public static final String STORAGE_DOWNLOAD_FILE = "/api/storage/{storedFileId}";
    public static final String STORAGE_INTERNAL_CREATE_FILE = "/internal/storage";
    public static final String STORAGE_GET_METADATA = "/api/storage/{storedFileId}/metadata";
}
